tsparser/src/parser/parser.rs

```diff
@@ -109,38 +111,132 @@ impl ParseContext {
 pub struct Parser<'a> {
     pc: &'a ParseContext,
     pass1: PassOneParser<'a>,
-
-    resources: Vec<Resource>,
-    binds: Vec<UnresolvedBind>,
 }
 
+#[derive(Debug)]
 pub struct ParseResult {
     pub resources: Vec<Resource>,
     pub binds: Vec<Lrc<Bind>>,
     pub usages: Vec<Usage>,
+    pub services: Vec<Service>,
 }
 
 impl<'a> Parser<'a> {
     pub fn new(pc: &'a ParseContext, pass1: PassOneParser<'a>) -> Self {
-        let resources = Vec::new();
-        let binds = Vec::new();
-        Self {
-            pc,
-            pass1,
-            resources,
-            binds,
-        }
+        Self { pc, pass1 }
     }
 
     /// Run the parser.
     pub fn parse(mut self) -> Result<ParseResult> {
-        for dir_root in &self.pc.dir_roots {
-            self.parse_root(dir_root)?;
+        fn ignored(entry: &walkdir::DirEntry) -> bool {
+            match entry.file_name().to_str().unwrap_or_default() {
+                "node_modules" | "encore.gen" => true,
+                x => x.starts_with('.'),
+            }
         }
-        self.inject_generated_service_clients();
 
-        let binds = resolve_binds(self.binds)?;
-        let resolver = UsageResolver::new(&self.pc.loader, &self.resources, &binds);
+        fn is_service(e: &walkdir::DirEntry) -> bool {
+            e.path().ends_with("encore.service.ts")
```

Comment:
very much a nit, but this would return true for filenames that ends with this, so a file called blaha.encore.service.ts would also return true (or is that what we want maybe?). Otherwise you could maybe use `e.path().file_name()' and check `.is_file()`

Reply:
ah wait, ends_with on path only checks whole path components, didnt know that!

Reply:
Yeah exactly :)